addon/hint/javascript-hint.js

```diff
@@ -144,12 +144,14 @@
         base = base[context.pop().string];
       if (base != null) gatherCompletions(base);
     } else {
-      // If not, just look in the global object and any local scope
+      // If not, just look in the global object, any local scope, and optional additional-context
       // (reading into JS mode internals to get at the local and global variables)
       for (var v = token.state.localVars; v; v = v.next) maybeAdd(v.name);
       for (var c = token.state.context; c; c = c.prev)
         for (var v = c.vars; v; v = v.next) maybeAdd(v.name)
       for (var v = token.state.globalVars; v; v = v.next) maybeAdd(v.name);
+      if (options && options.additionalContext !== undefined)
+        Object.keys(options.additionalContext).forEach(maybeAdd);
```

Comment:
The library still targets old runtimes that don't have `Object.keys`, so you'll need to use a `for`/`in` loop instead.

Also, for the line above, we use `!= null` for null checks throughout (instead of `!== undefined`).

Reply:
- [x] Converted `Object.keys` to `for (var keys in options.additionalContext) ...`
- [x] Converted `!==undefined` to `!=null`